hydra-tui/src/Hydra/TUI.hs

```diff
@@ -276,12 +275,12 @@ handleEvent client cardanoClient s = \case
   e ->
     continue $ s & warn ("unhandled event: " <> show e)
 
-sendInputAndTransition :: Client tx IO -> State -> ClientInput tx -> EventM n (Next State)
-sendInputAndTransition Client{sendInput} s input = case s ^? pendingL of
+sendInputAndTransition :: (a -> IO ()) -> State -> a -> EventM n (Next State)
+sendInputAndTransition sendAction s input = case s ^? pendingL of
   Just Pending -> do
     continue $ s & info "Transition already pending"
   Just NotPending -> do
-    liftIO $ sendInput input
+    liftIO $ sendAction input
```

Comment:
Should: generalize this differently

There is no point in passing `a` and `(a -> IO ())`, but instead an `IO ()` function would be better.

I suggest to rename also this function to make it clear that this is about doing an action and is not specific to `sendInput`

Reply:
In fact.. doing the action here does not make sense at all. Instead we could just convert this to a function named `setPending :: State -> State` and from the call sites we can do, e.g. `liftIO (externalCommit commitUTxO) >> continue s' & dialogStateL .~ NoDialog & setPending`